cmd/vmwrapper/vmwrapper.go

```diff
@@ -75,13 +75,42 @@ func main() {
 	// for cgroups that aren't managed by libvirt.
 	// If we don't do this, the VM pod will be killed by kubelet when Virtlet pod
 	// is removed dnd cgroup-per-qos is enabled in kubelet settings.
+	podId := os.Getenv(config.VmPodIdEnvVarName)
+	glog.V(5).Infof("Got VM POD ID: %s", podId )
+
+	computeCgroupRoot := os.Getenv(config.VmCgroupRootEnvVarName)
+	glog.V(5).Infof("Got CgroupRoot: %s", computeCgroupRoot )
+
 	cm := cgroups.NewManager(os.Getpid(), nil)
-	for _, ctl := range []string{"hugetlb", "systemd", "pids"} {
-		if _, err := cm.GetProcessController(ctl); err == nil {
-			err = cm.MoveProcess(ctl, "/")
+
+	controllers, err := cm.GetProcessControllers()
+
+	if err != nil {
+		glog.Warningf("Failed to get cgroup controllers: %v", err)
+	} else {
+		for ctl, ctlPath := range controllers {
+			glog.V(5).Infof("Cgroup controller name %s, path %s", ctl, ctlPath)
+
+			cgPath := "/"
+			// Preserve the previous behavior to put VM process to root for systemd, hugetlb and pids controllers
+			// otherwise, put the process under the desired cgroup created for the POD
+			if ctl != "systemd" && ctl != "hugetlb" && ctl != "pids" {
+				if controller, err := cm.GetProcessController(ctl); err == nil {
+					cgPath = computeCgroupRoot + "pod" + podId
```

Comment:
I remember in a previous PR you treat all VM pods as "guaranteed" QoS pods.  I don't why but it seems in K8S the crgoup of guaranteed pods are put under $controller\kubepods\burstable\podXXX group.  Should we also add "burstable" in the cgroup path for consistency?

Reply:
I noted that as well. the burstable, besteffort subgroup were there for containers but there is no guranteed subcg created under /kubepods/ at all. seems the Qos setting is missed. will take a look.

Reply:
@XiaoningDing Actually i misread your previous comment. the pod created with a guaranteed resource setting in the test, it is indeed under the /kubepods/ CG. the pasted output copy in the PR description is just hard to align.

However, this raised a flag to me for the previous implementation, in which uses a config string to set the ROOT parent and construct the CG parent with the PODID. This is actually incorrect. it provides the flexibility for runtime to "do some work" for the VM CG. however it carries drawbacks. Not only it "assumed" the guaranteed QoS for VM, but also it potentially splits the intention from the compute agent ( Arktos for example0 which CG parent the POD should be with).

I just pushed a change to use whatever CG parent the compute Agent set for the POD from the podSandBox configuration in the CRI. so that it leaves the total control of the compute agent for setting up the CG parent for the VM POD. this makes the design much straightforward.